File: Java/src/test/java/org/codecop/dependencies/b/MarketingCampaignTest.java
package org.codecop.dependencies.b;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import java.time.DayOfWeek;

import org.junit.Test;

public class MarketingCampaignTest {

    @Test
    public void tuesdayIsNotCrazy() {
        MarketingCampaign campaign = new MarketingCampaign() {
            @Override
            protected DayOfWeek dayOfWeek() {
                return DayOfWeek.TUESDAY;
            }
        };

        boolean isCrazySalesDay = campaign.isCrazySalesDay();

        assertFalse(isCrazySalesDay);
    }

    @Test
    public void fridayIsCrazy() {
        MarketingCampaign campaign = new MarketingCampaign() {
            @Override
            protected DayOfWeek dayOfWeek() {
                return DayOfWeek.FRIDAY;
            }
        };

        boolean isCrazySalesDay = campaign.isCrazySalesDay();

        assertTrue(isCrazySalesDay);
    }
}

File: Kotlin/src/test/kotlin/org/codecop/dependencies/b/MarketingCampaignTest.kt
package org.codecop.dependencies.b

import org.junit.Assert.assertFalse
import org.junit.Test

class MarketingCampaignTest {

    @Test
    fun test2() {
        val campaign = MarketingCampaign()

        val isCrazySalesDay = campaign.isCrazySalesDay()

        assertFalse(isCrazySalesDay)
    }

}